data/libs/functions/player.lua

```diff
@@ -554,3 +554,15 @@ function Player:calculateLootFactor(monster)
 		msgSuffix = suffix
 	}
 end
+
+function Player.setExhaustion(self, key, seconds)
+	return self:setStorageValue(key, os.time() + seconds)
+end
+
+function Player.getExhaustion(self, key)
+	return math.max(self:getStorageValue(key) - os.time(), 0)
+end
+
+function Player:hasExhaustion(self, key)
```

Comment:
luacheck caught this but: pay attention to how you define methods in lua, if you use `:` the argument `self` is automatic, only when you define it with `.` do you need to explicitly receive `self`

feedback: remove self from all 3 methods here and define them all with `:`